mixed/flink/flink-common/src/main/java/com/netease/arctic/flink/catalog/MixedCatalog.java

```diff
@@ -623,4 +651,131 @@ private void validateColumnOrder(CatalogBaseTable table) {
+    }
+
+    if (!(Objects.equal(ts1.getTableColumns(), ts2.getTableColumns())
+        && Objects.equal(ts1.getWatermarkSpecs(), ts2.getWatermarkSpecs())
+        && equalsPrimary)) {
+      throw new UnsupportedOperationException("Altering schema is not supported yet.");
+    }
+
+    if (!ct1.getPartitionKeys().equals(ct2.getPartitionKeys())) {
+      throw new UnsupportedOperationException("Altering partition keys is not supported yet.");
+    }
+  }
+
+  private void alterUnKeyedTable(UnkeyedTable table, CatalogBaseTable newTable) {
+    Map<String, String> oldProperties = table.properties();
+    Map<String, String> setProperties = Maps.newHashMap();
+
+    String setLocation = null;
+    String setSnapshotId = null;
+    String pickSnapshotId = null;
+
+    for (Map.Entry<String, String> entry : newTable.getOptions().entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+
+      if (Objects.equal(value, oldProperties.get(key))) {
+        continue;
+      }
+
+      if ("location".equalsIgnoreCase(key)) {
+        setLocation = value;
+      } else if ("current-snapshot-id".equalsIgnoreCase(key)) {
+        setSnapshotId = value;
+      } else if ("cherry-pick-snapshot-id".equalsIgnoreCase(key)) {
+        pickSnapshotId = value;
+      } else {
+        setProperties.put(key, value);
+      }
+    }
+
+    oldProperties
+        .keySet()
+        .forEach(
+            k -> {
+              if (!newTable.getOptions().containsKey(k)) {
+                setProperties.put(k, null);
+              }
+            });
+
+    FlinkAlterTableUtil.commitChanges(
+        table, setLocation, setSnapshotId, pickSnapshotId, setProperties);
+  }
+
+  private CatalogTable toCatalogTable(ArcticTable table, TableIdentifier tableIdentifier) {
+    Schema arcticSchema = table.schema();
+
+    Map<String, String> arcticProperties = Maps.newHashMap(table.properties());
+    fillTableProperties(arcticProperties);
+    fillTableMetaPropertiesIfLookupLike(arcticProperties, tableIdentifier);
+
+    List<String> partitionKeys = toPartitionKeys(table.spec(), table.schema());
+    return new CatalogTableImpl(
+        toSchema(arcticSchema, ArcticUtils.getPrimaryKeys(table), arcticProperties),
+        partitionKeys,
+        arcticProperties,
+        null);
+  }
+
+  private void alterKeyedTable(KeyedTable table, CatalogBaseTable newTable) {
+    Map<String, String> oldProperties = table.properties();
+    Map<String, String> setProperties = Maps.newHashMap();
+    for (Map.Entry<String, String> entry : newTable.getOptions().entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+      if (!Objects.equal(value, oldProperties.get(key))) {
+        setProperties.put(key, value);
+      }
+    }
+    oldProperties
+        .keySet()
+        .forEach(
+            k -> {
+              if (!newTable.getOptions().containsKey(k)) {
+                setProperties.put(k, null);
```

Comment:
Why setting a null value for the keys which doesn't exists in the newTable.Options?
```
tbl properties (
    k1 = v1
);
alter table tbl set k2=v2 where ....;
```
I think the k1-v1 property would be overrided after executing the ALTER statement.

Reply:
[FlinkAlterTableUtil.commitChanges](https://github.com/apache/iceberg/blob/main/flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/util/FlinkAlterTableUtil.java#L56) before commit will be remove